data_model/src/transaction.rs

```diff
@@ -331,16 +367,29 @@ impl TransactionValue {
         self.value.hash()
     }
 
-    /// [`Transaction`] payload.
+    /// Calculate transaction payload [`Hash`](`iroha_crypto::HashOf`).
+    #[cfg(feature = "std")]
+    #[cfg(feature = "transparent_api")]
+    pub fn hash_of_payload(&self) -> iroha_crypto::HashOf<TransactionPayload> {
+        self.value.hash_of_payload()
+    }
+
+    /// Return transaction instructions
     #[inline]
-    pub fn payload(&self) -> &TransactionPayload {
-        self.value.payload()
+    pub fn instructions(&self) -> &Executable {
+        self.value.instructions()
     }
 
-    /// [`iroha_crypto::SignatureOf`]<[`TransactionPayload`]>.
+    /// Creation timestamp
     #[inline]
-    pub fn signatures(&self) -> &SignaturesOf<TransactionPayload> {
-        self.value.signatures()
+    pub fn creation_time(&self) -> Duration {
+        self.value.creation_time()
+    }
+
+    /// Return transaction authority
+    #[inline]
+    pub fn authority(&self) -> &AccountId {
+        self.value.authority()
     }
 }
```

Comment:
we should consider removing all methods defined on `TransactionValue`. I'm not sure what desired api would be here

Reply:
we can remove all methods on `TransactionValue` and leave only `fn error(&self)` and implement `AsRef<SignedTransaction>`

Reply:
What do you mean by that `fn error(&self)` ?